client/asset/btc/btc.go

```diff
@@ -4325,90 +4325,103 @@ func (btc *intermediaryWallet) prepareRedemptionRequestsForBlockCheck() []*findR
 // and a goroutine is started to check if any contracts in the
 // findRedemptionQueue are redeemed in the new blocks.
 func (btc *intermediaryWallet) reportNewTip(ctx context.Context, newTip *block) {
+	// Lock to avoid concurrent reportNewTip execution for simplicity.
 	btc.tipMtx.Lock()
 	defer btc.tipMtx.Unlock()
 
 	prevTip := btc.currentTip
 	btc.currentTip = newTip
+
 	btc.log.Debugf("tip change: %d (%s) => %d (%s)", prevTip.height, prevTip.hash, newTip.height, newTip.hash)
+
 	go btc.tipChange(nil)
 
 	reqs := btc.prepareRedemptionRequestsForBlockCheck()
+	if len(reqs) == 0 {
+		return
+	}
+
 	// Redemption search would be compromised if the starting point cannot
 	// be determined, as searching just the new tip might result in blocks
 	// being omitted from the search operation. If that happens, cancel all
 	// find redemption requests in queue.
 	notifyFatalFindRedemptionError := func(s string, a ...interface{}) {
-		for _, req := range reqs {
-			req.fail("tipChange handler - "+s, a...)
-		}
+		btc.fatalFindRedemptionsError(fmt.Errorf("tipChange handler - "+s, a...), reqs)
 	}
 
-	var startPoint *block
+	startHash := &newTip.hash
+
 	// Check if the previous tip is still part of the mainchain (prevTip confs >= 0).
 	// Redemption search would typically resume from prevTipHeight + 1 unless the
+	// Redemption search would typically resume from prevTip.height + 1 unless the
```

Comment:
Ah, yeah I was adjusting this comment to replace **prevTipHeight** -> **prevTip.height** (to remove any potential ambiguity), removed the duplicate part now: https://github.com/decred/dcrdex/pull/2227/commits/4d93b683a6b01ece8f20df3be72cbd55ff784a30.

Reply:
Maybe I'm missing something then, but case that ensured it resumes from `prevTip.height + 1` even if "2 or more blocks mined since last tip" is now removed.  That's unrelated to reorg.

Reply:
> but the case that ensured it resumes from prevTip.height + 1 even if "2 or more blocks mined since last tip" is now removed.

Currently it resumes from `prevTip.height` indeed! Should start with `startHash` +1 actually (so the comment is right), let me see how we can handle that.

Reply:
The code under the removed `case newTip.height-prevTipHeader.Height > 1:` is basically what we are after.

Reply:
Adjusted now to fix this: https://github.com/decred/dcrdex/pull/2227/commits/9b7641d9f3d2dba9ce27cffce126ab9a444f0a99.